hist/histpainter/src/THistPainter.cxx

```diff
@@ -3739,42 +3739,43 @@ char *THistPainter::GetObjectInfo(Int_t px, Int_t py) const
    if (fH->GetDimension() == 1) {
       if (fH->InheritsFrom(TProfile::Class())) {
          TProfile *tp = (TProfile*)fH;
-         snprintf(info,200,"(x=%g, y=%g, binx=%d, binc=%g, bine=%g, binn=%d)",
+         info = TString::Format("(x=%g, y=%g, binx=%d, binc=%g, bine=%g, binn=%d)",
                   x, y, binx, fH->GetBinContent(binx), fH->GetBinError(binx),
                   (Int_t) tp->GetBinEntries(binx));
       }
       else {
          Double_t integ = 0;
          for (Int_t bin=binmin;bin<=binx;bin++) {integ += fH->GetBinContent(bin);}
-         snprintf(info,200,"(x=%g, y=%g, binx=%d, binc=%g, Sum=%g)",
+         info = TString::Format("(x=%g, y=%g, binx=%d, binc=%g, Sum=%g)",
                   x,y,binx,fH->GetBinContent(binx),integ);
       }
    } else if (fH->GetDimension() == 2) {
       if (fH->InheritsFrom(TH2Poly::Class())) {
          TH2Poly *th2 = (TH2Poly*)fH;
          biny = th2->FindBin(x,y);
-         snprintf(info,200,"%s (x=%g, y=%g, bin=%d, binc=%g)",
+         info = TString::Format("%s (x=%g, y=%g, bin=%d, binc=%g)",
```

Comment:
One could use info.Form(...) instead of info = TString::Format(...) and in most lines afterwards